networks/chainconfig.go

```diff
@@ -0,0 +1,141 @@
+package networks
+
+import (
+	_ "embed"
+	"fmt"
+	"slices"
+	"sort"
+	"sync"
+
+	registry "github.com/pinax-network/graph-networks-libs/packages/golang/lib"
+)
+
+//go:embed TheGraphNetworksRegistry.json
+var embeddedRegistryJSON []byte
+
+// Package networks provides access to The Graph's network registry and helpers.
+
+// NetworkRegistry is a thin wrapper around a [map[string]*registry.Network] to add some helper methods.
+type NetworkRegistry map[string]*registry.Network
+
+var (
+	registryNetworks     NetworkRegistry
+	registryNetworksOnce sync.Once
+)
+
+// getRegistryNetworks fetches and caches all networks from the registry (no filtering).
+func getRegistryNetworks() NetworkRegistry {
+	registryNetworksOnce.Do(func() {
+		reg, err := registry.FromLatestVersion()
+		if err != nil {
+			// Fallback: use embedded JSON
+			reg, err = registry.FromJSON(embeddedRegistryJSON)
+			if err != nil {
+				panic(fmt.Sprintf("Failed to load registry from both network and embedded JSON: %v", err))
+			}
+		}
+		m := make(map[string]*registry.Network)
+		for i, net := range reg.Networks {
+			m[net.ID] = &reg.Networks[i]
+		}
+		registryNetworks = m
+
+		// Add custom Tron network until it's added to the registry
+		registryNetworks.addCustomNetwork(&registry.Network{
```

Comment:
Not sure if all the data is valid here, I believe so?

Reply:
Seems quite good, I will probably task you to open a PR for this on the network registry

Reply:
For the PR for network registry, do I need to add more stuff? How do I gather the missing stuff?

Reply:
List me the missing pieces and I'll ask TRON team